tardis/tests/tests_slow/conftest.py

```diff
@@ -50,16 +49,13 @@ def plot_object(request):
     return PlotUploader(request)
 
 
-def setup_names():
-    slow_tests_dir = os.path.dirname(os.path.realpath(__file__))
-    names = list()
-    for child in os.listdir(slow_tests_dir):
-        if os.path.isdir(os.path.join(slow_tests_dir, child)):
-            names.append(child)
-    return names
+setup_names = [
+    name for name in os.listdir(os.path.dirname(os.path.realpath(__file__)))
+    if os.path.isdir(os.path.join(os.path.dirname(os.path.realpath(__file__)), name))
+]
 
 
-@pytest.fixture(scope="class", params=setup_names())
+@pytest.fixture(scope="class", params=setup_names)
```

Comment:
I did not declare `setup_names` here because:
![image](https://cloud.githubusercontent.com/assets/10494087/16412219/a9e6df18-3d48-11e6-886c-4f71872f9bf1.png)
This is unusual, there aren't many alternatives to try out !